pages/review/[lesson].tsx

```diff
@@ -43,6 +43,10 @@ const Review: React.FC<QueryDataProps<AppData>> = ({ queryData }) => {
     router.push('/login')
     return null
   }
+  if (!lessons.map((l: Lesson) => l.id.toString()).includes(currentlessonId)) {
```

Comment:
Can we use a better variable name than `l`?

Is the `toString` needed too? In our typedefs, we define the `id` field as a string, so it should be a string already
https://github.com/garageScript/c0d3-app/blob/master/graphql/typeDefs.ts#L130

Reply:
`if (!lessons.map((l: Lesson) => l.id.toString()).includes(currentlessonId))`

I don't think the line above is doing what you want it to. `map` function returns an array, so it's never going to be truthy, it can only be falsy.

This use of the `find` function should return `undefined` if the lessonId is not found: 
```
lessons.find((lesson)=>lesson.id === currentlessonId)
```

Reply:
I redirected properly when I tested locally. First map returned an array and then includes checked if currentlessonId is in this array. But I agree `!lessons.find((lesson: Lesson) => lesson.id.toString() === currentlessonId)` is probably more readable. Updated.

Reply:
`l.id.toString())` <-- ah, thanks for explaining! i indeed read the function wrong because I didnt notice this extra `)`